pkg/kubelet/kubelet.go

```diff
@@ -666,6 +667,8 @@ type Kubelet struct {
 	reservation kubetypes.Reservation
 }
 
+var _ stats.KubeletStatsProvider = &Kubelet{}
```

Comment:
(nit) Currently `Kubelet` gets passed as a `server.HostInterface`, maybe it would be better to assert that here, and assert that `server.HostInterface` is a `stats.KubeletStatsProvider`?